drivers/xbee/xbee.c

```diff
@@ -452,24 +448,10 @@ int xbee_init(xbee_t *dev, uart_t uart, uint32_t baudrate,
     /* exit command mode */
     _at_cmd(dev, "ATCN\r");
 
-    /* set default short address, use CPU ID if available */
-#if CPUID_ID_LEN
-    /* get CPU ID */
-    uint8_t id[CPUID_ID_LEN];
-    cpuid_get(id);
-    /* set address */
-    memset(dev->addr_short, 0, 2);
-    for (int i = 0; i < CPUID_ID_LEN; i++) {
-        dev->addr_short[i & 0x01] ^= id[i];
-    }
-#else
-    dev->addr_short[0] = (uint8_t)(XBEE_DEFAULT_SHORT_ADDR >> 8);
-    dev->addr_short[1] = (uint8_t)(XBEE_DEFAULT_SHORT_ADDR);
-#endif
-    _set_addr(dev, dev->addr_short, 2);
     /* load long address (we can not set it, its read only for Xbee devices) */
-    _get_addr_long(dev, dev->addr_long, 8);
+    _get_addr_long(dev, dev->addr_long.uint8, 8);
     /* set default channel */
+    _set_addr(dev, &((dev->addr_long).uint8[7]), 2);
```

Comment:
I think you mean `&((dev->addr_long).uint8[`**6**`])` here, as you copy byte 7 and 8!?